client/alloc_runner.go

```diff
@@ -643,10 +644,18 @@ func (r *AllocRunner) setTaskState(taskName, state string, event *structs.TaskEv
 		r.appendTaskEvent(taskState, event)
 	}
 
-	if state == "" {
+	if lazySync {
```

Comment:
No because we want to push the events. The lazy sync is useful when the event will be immediately followed up so that you can avoid a networked sync.